### PR TITLE
Fixes #3058 : Make the requests faster and less network intensive by using fields parameter

### LINKED ISSUE #3058: Make the requests faster and less network intensive by using the ?fields= parameter
Currently we send the following query when clicking on a Brand: `https://world.openfoodfacts.org/brand/Veuve%20Clicquot/1.json`  **(6.66 KB for 1 product)**

We should ask instead:
`https://world.openfoodfacts.org/brand/Veuve%20Clicquot/1.json?fields=brands,product_name_fr,product_name,image_small_url,image_front_thumb_url,quantity,nutrition_grades_tags` **(417 Bytes for 1 product)**

If you multiply the difference by a larger number of products (as is often the case, here there's a spelling mistake on the brand, thus only 1 product), you get a **much larger size difference** that's at least (6,2KB * N) (actually more, since there are no ingredients not nutrients on this champagne).
This might not matter on 5G or 4G, but it does matter on EDGE, 3G, and when you're in a supermarket with very bad network.

**Notes:**
- I added both image_small_url,image_front_thumb_url but in truth we can keep only the one we actually use.
- I added both product_name_XX and product_name to be able to display the product name in the language of the app, if available.

This should be done for  **Products to be completed** and **Facets** (the pages that appear when you click a value on a product page). Search and Scan have already been optimized

### PR DESCRIPTION
**Pull request by Varsha-29**
Added query fields to speed up brand detail fetch process
Changes made in OpenFoodAPIClient and OpenFoodAPIService

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java [hunk 1]
```diff
@@ -346,8 +346,8 @@ public OpenFoodAPIService getAPIService() {
     }
 
     public void getBrand(final String brand, final int page, final OnBrandCallback onBrandCallback) {
-
-        apiService.getProductByBrands(brand, page).enqueue(new Callback<Search>() {
+        String fields = "brands,product_name_fr,product_name,image_small_url,image_front_thumb_url,quantity,nutrition_grades_tags";
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java:349]: Make it final.
```suggestion
        final String fields = "brands,product_name_fr,product_name,image_small_url,image_front_thumb_url,quantity,nutrition_grades_tags";
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java [hunk 2]
```diff
@@ -346,8 +346,8 @@ public OpenFoodAPIService getAPIService() {
...
+        apiService.getProductByBrands(brand, page, fields).enqueue(new Callback<Search>() {
             @Override
             public void onResponse(Call<Search> call, Response<Search> response) {
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java:352]:
```suggestion
            public void onResponse(@NonNull Call<Search> call, @NonNull Response<Search> response) {
```

**Comment by Varsha-29** [comment]
Hi @VaiTon , 
Thanks for reviewing it. I was not quite clear about the other facets. Could you list them?

**Comment by VaiTon** [comment]
We need to see which fields we are using, so we need to check every call to the function i think.
@teolemon what do you think?

**Comment by teolemon** [comment]
Normally, the facets show all exactly the same values, which are the ones listed for brands. @VaiTon @Varsha-29

**Comment by teolemon** [comment]
product_name_fr should be product_name_XX where XX is the locale of the app

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java [hunk 3]
```diff
@@ -87,6 +87,7 @@
     private static OkHttpClient httpClient = Utils.HttpClientBuilder();
     private final OpenFoodAPIService apiService;
     private Context mActivity;
+    private final String fieldsToFetchProductDetail = "brands,product_name_fr,product_name,image_small_url,image_front_thumb_url,quantity,nutrition_grades_tags";
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java:90]: It is a constant, so i would rename it to:
```suggestion
    private final String FIELDS_TO_FETCH_FACETS = "brands,product_name_fr,product_name,image_small_url,image_front_thumb_url,quantity,nutrition_grades_tags";
```

Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIClient.java:90]: Also what @teolemon said, `product_name_fr` should be `product_name_xx` with `xx` being the phone locale.

**Comment by VaiTon** [comment]
Thank you!

**Comment by teolemon** [comment]
I'm merging. Congratulations on your first OFF PR @Varsha-29 🎉

**Pull request merged**